journal.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""A Pyramid app for my learning journal
+
+"""
+from __future__ import unicode_literals
+import os
+from pyramid.config import Configurator
+from pyramid.view import view_config
+from waitress import serve
+import sqlalchemy as sa
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import scoped_session, sessionmaker
+from zope.sqlalchemy import ZopeTransactionExtension
+import datetime
+from pyramid.httpexceptions import HTTPFound
+from sqlalchemy.exc import DBAPIError
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from cryptacular.bcrypt import BCRYPTPasswordManager
+from pyramid.security import remember, forget
+
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension()))
+DATABASE_URL = os.environ.get(
+    'DATABASE_URL',
+    'postgresql://jesse:Jjk5646!@localhost:5432/learning-journal'
+)
+
+
+Base = declarative_base()
+engine = sa.create_engine(DATABASE_URL)
+# Session = sessionmaker(bind=engine)
+
+
+class Entry(Base):
+    """Make a new entry
+    """
+    __tablename__ = 'entries'
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+    title = sa.Column(sa.Unicode(127), nullable=False)
+    date = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
+    )
+    content = sa.Column(sa.UnicodeText, nullable=False)
+
+    @classmethod
+    def write(cls, title=None, content=None, session=None):
+        if session is None:
+            session = DBSession
+        instance = cls(title=title, content=content)
+        session.add(instance)
+        return instance
+
+    @classmethod
+    def all(cls, session=None):
+        if session is None:
+            session = DBSession
+        return session.query(cls).order_by(cls.date.desc()).all()
+
+
+def init_db():
+    """Make a new entries table
+    """
+    Base.metadata.create_all(bind=engine)
```

Comment:
Where does `engine` come from?  You really don't want that to be at module scope.